### PR TITLE
fix issue 256, 257

### PR DESCRIPTION
**Pull request by webelf101**

### CODE REVIEW

File: solidity/contracts/Gravity.sol [hunk 1]
```diff
@@ -230,14 +228,12 @@ contract Gravity is ReentrancyGuard {
 		);
 
 		// Check that new validators and powers set is well-formed
-		require(_newValidators.length == _newPowers.length, "Malformed new validator set");
+		require(_newValidators.length == _newPowers.length || _newValidators.length == 0, "Malformed new validator set");
```
Comment (jtremback) [inline solidity/contracts/Gravity.sol:231]: This is the first substantive change in this PR (other than the signature struct refactor). I guess it doesn't hurt but it's also not very useful. This entire require statement could probably go, and probably should. I just never got around to removing it.

File: solidity/contracts/Gravity.sol [hunk 2]
```diff
@@ -252,16 +248,26 @@ contract Gravity is ReentrancyGuard {
...
+			cumulativePower = cumulativePower + _newPowers[i];
+			if (cumulativePower > state_powerThreshold) {
+				break;
+			}
+		}
+		require(
+			cumulativePower > state_powerThreshold,
+			"Submitted validator set signatures do not have enough power."
+		);
+
```
Comment (jtremback) [inline solidity/contracts/Gravity.sol:262]: This is not useful IMO. You are only protecting the validators from themselves. This logic should be, and is, on the cosmos side.

**Review by jtremback — CHANGES_REQUESTED** [review]
The refactor to use the signature struct is good. We use parallel arrays because at the time thats all that solidity supported in our use case.

The other changes, which i have commented on, are a waste of gas. If you would like i can get on a call and explain the threat model and why these changes are not neccesary. We could probably go even further and remove some existing requires. I'm happy to help guide that effort.

File: solidity/contracts/Gravity.sol (hunk 1 above)
Comment (webelf101) [reply solidity/contracts/Gravity.sol:231]: This update is related to the issue https://github.com/PeggyJV/gravity-bridge/issues/257
Should I ignore this issue? @jtremback

File: solidity/contracts/Gravity.sol (hunk 2 above)
Comment (webelf101) [reply solidity/contracts/Gravity.sol:262]: This is also related to the issue https://github.com/PeggyJV/gravity-bridge/issues/257

**Pull request merged**